Update python requirements

Patater opened the pull request:
### Summary of changes <!-- Required -->

Update Python dependencies to newer versions to get important updates for them.

#### Impact of changes <!-- Optional -->
Not yet known.

#### Migration actions required <!-- Optional -->
Not yet known.

### Documentation <!-- Required -->

None

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [X] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

Patater committed:
python: Update requirements

Update versions of urllib3, PyYAML, jinja2, cryptography.

jeromecoutant commented on requirements.txt:
```diff
@@ -1,16 +1,16 @@
 colorama==0.3.9
-urllib3[secure]==1.24.2
+urllib3[secure]>=1.26.5
 prettytable==0.7.2
```
It is not linked with this PR :-)
But I often have issues with that prettytable requirement...
It is not in line with pyOCD: 'prettytable>=2.0,<3.0',
https://github.com/pyocd/pyOCD/blob/master/setup.py#L74

Maybe we can relax the version number here ?
Thx

Patater replied:
We need the higher version to get security fixes, so we can't relax. We can update pyOCD.

jeromecoutant replied:
pyOCD requests a higher version...

Patater replied:
OK. This should work fine then. We are going to use `>=` rather than `==`.

Patater replied:
I followed up on our CI failure with the newer prettytable. We had pinned mbed-os-tools's requirement on prettytable as documented in a previous commit:

If we want to update prettytable, we need to drop Python 2.7 and 3.5 support.
```
    Latest release of `PrettyTable` (2.0.0) removed support for Python 2.7
    and 3.5 and is causing CI failures. We are still supporting those
    versions in mbed-os-tools. Therefore pin `PrettyTable` to 1.0.1.
```

LDong-Arm replied:
If we drop Python 2.7 and 3.5 (which are deprecated), could we also add `cysecuretools`? This package is needed by CYTFM_064B0S2_4343W (and one or two other Cypress targets if I remember correctly). We didn't add it because it only supports Python 3.6 and above.

0xc0170 replied:
We should be able to drop 2.7 and 3.5 (I thought 2.7 was already no supported).

Patater replied:
Dropping python2.7 will involve coordinated changes to many repos (mbed-os-tools, icetea, mbed-ls. I'll file an issue for doing this separately, including updating prettytable.

rwalton-arm commented on tools/cmake/requirements.txt:
```diff
@@ -2,3 +2,5 @@ prettytable==0.7.2
 Jinja2>=2.10.1,<2.11
 intelhex>=2.3.0,<3.0.0
+cbor>=1.0.0
+Click>=7.0,<7.1
```
Why set the upper bound to Click 7.1? Why not 8? Seems like we're setting ourselves up for more dependency conflicts down the line by over-specifying this. If we do require this particular range for some reason, please explain why in the commit message.

Patater replied:
Sorry, this was from a commit meant for https://github.com/ARMmbed/mbed-os/pull/14643. Removed from this PR

Patater committed:
travis: Use latest pip

As we depend on the cryptography library, and the pip version that comes
with Ubuntu Focal isn't new enough to avoid the following error about a
rust compiler and pre-built wheels, ensure Travis uses an up-to-date
pip.

    error: can't find Rust compiler

    If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.

    To update pip, run:

        pip install --upgrade pip

Patater committed:
travis: Drop testing Python 3.5 and 3.6

A few libraries we depend on no longer support old versions of Python
like 3.5 and 3.6. Remove testing for these old versions.

Patater committed:
travis: Remove trailing whitespace
